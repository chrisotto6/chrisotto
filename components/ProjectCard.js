import Image from './Image'
import Link from 'next/link'
import { FaGithub, FaExternalLinkSquareAlt } from 'react-icons/fa'

const ProjectCard = ({ title, description, img, githubUrl, projectUrl, tech, dataTestid }) => (
  <div
    className="max-w-xs overflow-hidden rounded-xl shadow-lg dark:bg-gray-600 dark:shadow-cyan-600/50"
    data-testid={dataTestid}
  >
    <Image className="w-full" alt={title} src={img} width={544} height={306} />
    <div className="px-6 py-4">
      <div className="mb-2 text-xl font-bold">{title}</div>
      <p className="text-base text-gray-500 dark:text-gray-100">{description}</p>
    </div>
    <div className="flex px-6 pt-3 pb-2">
      {tech.map((t) => (
        <span
          key={t}
          className="mr-2 mb-2 inline-block rounded-full bg-gray-200 px-3 py-1 text-sm font-semibold text-gray-500 dark:bg-cyan-600 dark:text-gray-100"
        >
          {t}
        </span>
      ))}
    </div>
    <div className="mb-3 flex space-x-4 px-6 pt-3 pb-3">
      <Link href={githubUrl}>
        <FaGithub size={30} />
      </Link>
      <Link href={projectUrl}>
        <FaExternalLinkSquareAlt size={30} />
      </Link>
    </div>
  </div>
)

export default ProjectCard
