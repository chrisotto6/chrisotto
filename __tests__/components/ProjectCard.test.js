import { render } from '@testing-library/react'
import ProjectCard from '@/components/ProjectCard'

it('renders ProjectCard correctly', () => {
  const { container } = render(
    <ProjectCard
      title="test"
      description="test"
      img="/test"
      tech={['node', 'graphql']}
      githubUrl="www.github.com"
      projectUrl="www.google.com"
    />
  )
  expect(container).toMatchSnapshot()
})
