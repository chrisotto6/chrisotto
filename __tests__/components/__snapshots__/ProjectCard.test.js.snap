// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProjectCard correctly 1`] = `
<div>
  <div
    class="max-w-xs overflow-hidden rounded-xl shadow-lg dark:bg-gray-600 dark:shadow-cyan-600/50"
  >
    <img
      alt="test"
      class="w-full"
      data-nimg="1"
      decoding="async"
      height="306"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Ftest&w=3840&q=75"
      srcset="/_next/image?url=%2Ftest&w=640&q=75 640w, /_next/image?url=%2Ftest&w=750&q=75 750w, /_next/image?url=%2Ftest&w=828&q=75 828w, /_next/image?url=%2Ftest&w=1080&q=75 1080w, /_next/image?url=%2Ftest&w=1200&q=75 1200w, /_next/image?url=%2Ftest&w=1920&q=75 1920w, /_next/image?url=%2Ftest&w=2048&q=75 2048w, /_next/image?url=%2Ftest&w=3840&q=75 3840w"
      style="color: transparent; max-width: 100%; height: auto;"
      width="544"
    />
    <div
      class="px-6 py-4"
    >
      <div
        class="mb-2 text-xl font-bold"
      >
        test
      </div>
      <p
        class="text-base text-gray-500 dark:text-gray-100"
      >
        test
      </p>
    </div>
    <div
      class="flex px-6 pt-3 pb-2"
    >
      <span
        class="mr-2 mb-2 inline-block rounded-full bg-gray-200 px-3 py-1 text-sm font-semibold text-gray-500 dark:bg-cyan-600 dark:text-gray-100"
      >
        node
      </span>
      <span
        class="mr-2 mb-2 inline-block rounded-full bg-gray-200 px-3 py-1 text-sm font-semibold text-gray-500 dark:bg-cyan-600 dark:text-gray-100"
      >
        graphql
      </span>
    </div>
    <div
      class="mb-3 flex space-x-4 px-6 pt-3 pb-3"
    >
      <a
        href="www.github.com"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 496 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
      <a
        href="www.google.com"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
